packages/inter-protocol/src/vpool-xyk-amm/doublePool.js

```diff
@@ -57,15 +58,30 @@ export const makeDoublePool = (
     const inPoolSeat = collateralInPool.getPoolSeat();
     const outPoolSeat = collateralOutPool.getPoolSeat();
 
-    seat.decrementBy(harden({ In: prices.swapperGives }));
-    inPoolSeat.decrementBy(harden({ Central: prices.inPoolDecrement }));
-    outPoolSeat.decrementBy(harden({ Secondary: prices.outPoolDecrement }));
-    seat.incrementBy(harden({ Out: prices.swapperGets }));
-    inPoolSeat.incrementBy(harden({ Secondary: prices.inPoolIncrement }));
-    outPoolSeat.incrementBy(harden({ Central: prices.outPoolIncrement }));
-    feeSeat.incrementBy(harden({ Fee: prices.protocolFee }));
+    atomicTransfer(
+      zcf,
+      harden([
+        [seat, seat, { In: prices.swapperGives }, { Out: prices.swapperGets }],
+        [
+          inPoolSeat,
+          inPoolSeat,
+          { Central: prices.inPoolDecrement },
+          { Secondary: prices.inPoolIncrement },
+        ],
+        [
+          outPoolSeat,
+          outPoolSeat,
+          { Secondary: prices.outPoolDecrement },
+          { Central: prices.outPoolIncrement },
+        ],
```

Comment:
Done, but somewhat differently. PTAL.

Reply:
This is not the only case in which the individual lines don't balance. The situation where `[seat1, seat2, keyedAmount1, keyedAmount2]` makes sense to me is when the amounts are the same and the keys differ. If the amounts are different I'd prefer to see it as two distinct lines, as you did with `doublePool`. 

I was hoping for more general support for the other patterns, since there's so much noise and redundancy here. The distinct cases I see are
```
atomicSwap(zcf, {
   xfer: [seat1, seat2, { key: amount }],
   xfer: [seat1, seat2, key1, key2, amount],
   from: [seat3, key3, amount3],
   to: [seat4, key4, amount4],
});
```

The labels are clearly wrong, but this seems like it could be significantly more concise. The first two vastly predominate, which is all the more reason to make the second two visually distinct. I don't have a concrete proposal, just a suggested direction.

You're right that making this change isn't necessary. I'll point out other cases where the model doesn't seem a good fit to me.

Reply:
I agree with @Chris-Hibbert this could use more tweaking before landing.

Some things to try to satisfy:
- easy to re-use a {keyword: amount} object because they're often already defined
- not repeating the keyword or amount unnecessarily
- not having to bracket the array in the single transfer case
- not having `undefined` in the arguments


**single transfer on the same key**
```js
atomicTransfer(zcf, fromSeat, toSeat, { keyword: amount });
```

For other cases, @Chris-Hibbert and @turadg propose having separate functions.

**single transfer to a different key**
```js
// in this PR
atomicTransfer(zcf, fromSeat, toSeat, { sourceKeyword: amount }, { targetKeyword: amount });
// dedupe amount with same function
atomicTransfer(zcf, fromSeat, toSeat, { sourceKeyword: amount }, targetKeyword);
// new function for new signature
atomicTransferRename(zcf, fromSeat, toSeat, sourceKeyword, targetKeyword, amount );
```

**rearrangement**
codify the pattern of splitting "from" and "to", removing the need for `undefined` 
```js
atomicRearrangement(zcf, {
  from: [
    [seat, { In: prices.swapperGives }],
    [inPoolSeat, { Central: prices.inPoolDecrement }],
    [outPoolSeat, { Secondary: prices.outPoolDecrement }],
  ],
  to: [
    [seat, { Out: prices.swapperGets }],
    [inPoolSeat, { Secondary: prices.inPoolIncrement }],
    [outPoolSeat, { Central: prices.outPoolIncrement }],
    [feeSeat, { Fee: prices.protocolFee }],
  ],
});

```

Reply:
These are great suggestions. I took big steps in these directions, but somewhat differently. PTAL and let me know what you think.

- The thing that used to be called `atomicTransfer` I renamed `atomicRearrange`.
- I then used `atomicTransfer` as you suggest above, for the special case of a single oneway transfer. I was surprised that this was the majority of cases: 22 `atomicTransfer(` vs 18 remaining `atomicRearrange`.
- for expressing from-only portions separately from to-only portions, I created to helpers `fromOnly` and `toOnly`. Both are mostly used only in `doublePool.js`. See if your pleased by how it looks there.

Reply:
I now understand @Chris-Hibbert 's criticisms about using one TransferArgs line with separate fromAmounts and toAmounts where the amounts are different. I think I took care of all of those, turning each into separate fromOnly and toOnly lines.

This though does suggest and error check: atomicTransfer could enforce that when a single TransferArgs line has both fromAmounts and toAmounts, that they are have the same total amounts. This ensures that each line either
   * `fromOnly` - takes only from one side
   * `toOnly` - gives only to one side
   * makes a balanced oneway transfer from one seat to another, obeying conservation by itself

Does this error check seem like a good idea? With this error check in place, what would you think of the overall API?

Reply:
I do want to avoid notation like `[seat1, seat2, key1, key2, amount],` because an AmountKeywordRecord can have multiple `keyword: amount` properties. It should be taken as an aggregate of all those amounts. And sometimes what we have coming in is the AmountKeywordRecord, which we should not need to take apart.

With the API in the current PR, there is still a bit of repetition left. Without the error check above (which I have not yet implemented), I can see the hazard. But with that error check, I think the repetition is harmless and actually makes things clearer. WDYT?

Reply:
> an AmountKeywordRecord can have multiple keyword: amount properties. It should be taken as an aggregate of all those amounts. And sometimes what we have coming in is the AmountKeywordRecord, which we should not need to take apart.

That's a good argument. I expect the error check to be sufficient. The repetition I was trying to avoid is an annoyance, but in the face of existing AKRs is acceptable.

Looking...

Reply:
Per-transferPart error check now implemented. In CI now, so we'll see if any current contracts violate it.

Done.